examples/custom-server/server.js

```diff
@@ -0,0 +1,33 @@
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+const accepts = require('accepts')
+
+const app = next({ dev: true })
+const handle = app.getRequestHandler()
+
+app.prepare()
+.then(() => {
+  createServer((req, res) => {
+    const accept = accepts(req)
+    const type = accept.type(['json', 'html'])
+    if (type === 'json') {
+      handle(req, res)
+      return
+    }
+
+    const { pathname, query } = parse(req.url, true)
```

Comment:
`handle(req, res, pathname, query)` seems better since it's same arguments with `app.render()`.
But on `handle()`, we don't want to allow customizing `pathname` and `query` like `app.render()`, so un-destructured url object also makes sense, umm.